pyserini/index/lucene/_base.py

```diff
@@ -535,3 +537,36 @@ def stats(self) -> Dict[str, int]:
             index_stats_dict[term] = index_stats_map.get(term)
 
         return index_stats_dict
+
+    def dump_documents_BM25(self, file_path="collections/documents_bm25_dump.json"):
+        """Dumps out all the document vectors with BM25 weights in Pyserini's JSON vector format 
+
+        Parameters
+        ----------
+        file_path : str
+            file path to dump JSON file
+        """
+        first_iter = True
+        output_string =  "[\n"
+
+        for i in tqdm(range(self.stats()["documents"])):
```

Comment:
defensive programming here. check 'documents' first to make sure it's not None, etc.

Reply:
Does it look good now?`